omod/src/test/java/org/openmrs/module/idgen/web/controller/IdentifierSourceRestControllerTest.java

```diff
@@ -0,0 +1,240 @@
+        identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+        identifierSource.add("firstIdentifierBase", "1");
+        identifierSource.add("prefix", "ABC");
+        identifierSource.add("suffix", "XYZ");
+        identifierSource.add("minLength", "36");
+        identifierSource.add("maxLength", "38");
+        identifierSource.add("baseCharacterSet", "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
+        identifierSource.add("sourceType", SEQUENTIAL_IDENTIFIER_SOURCE_TYPE);
+
+        String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+
+        MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+
+        SimpleObject newIdentifierSource = deserialize(handle(req));
+
+        SequentialIdentifierGenerator generatedIdentifierSource = (SequentialIdentifierGenerator) service.getAllIdentifierSources(false).get(7);
+
+        assertEquals(PropertyUtils.getProperty(newIdentifierSource, "uuid"), generatedIdentifierSource.getUuid());
+        assertEquals(generatedIdentifierSource.getName(), "test sequential identifier");
+        assertEquals(generatedIdentifierSource.getDescription(), "test sequential identifier");
+        assertEquals(generatedIdentifierSource.getFirstIdentifierBase(), "1");
+        assertEquals(generatedIdentifierSource.getPrefix(), "ABC");
+        assertEquals(generatedIdentifierSource.getSuffix(), "XYZ");
+        assertEquals(generatedIdentifierSource.getMinLength().toString(), "36");
+        assertEquals(generatedIdentifierSource.getMaxLength().toString(), "38");
+        assertEquals(generatedIdentifierSource.getBaseCharacterSet(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
+        assertEquals(generatedIdentifierSource.getIdentifierType().getUuid(), IDENTIFIER_TYPE_UUID);
+        assertEquals(initialIdentifierSourceCount + 1, getAllCount());
+
+    }
+
+    @Test
+    public void shouldSaveARemoteIdentifierGenerator() throws Exception {
+
+        long initialIdentifierSourceCount = getAllCount();
+
+        SimpleObject identifierSource = new SimpleObject();
+        identifierSource.add("name", "test remote identifier");
+        identifierSource.add("description", "test remote identifier");
+        identifierSource.add("url", "http://localhost");
+        identifierSource.add("username", "openmrsuser");
+        identifierSource.add("password", "openmrspassword");
+        identifierSource.add("identifierType", IDENTIFIER_TYPE_UUID);
+        identifierSource.add("sourceType", REMOTE_IDENTIFIER_SOURCE_TYPE);
+
+        String jsonIdentifierSource = new ObjectMapper().writeValueAsString(identifierSource);
+
+        MockHttpServletRequest req = newPostRequest(getURI(), jsonIdentifierSource);
+
+        SimpleObject newIdentifierSource = deserialize(handle(req));
+
+        RemoteIdentifierSource generatedIdentifierSource = (RemoteIdentifierSource) service.getAllIdentifierSources(false).get(7);
```

Comment:
Why do you fetch all identifier sources instead of the one you are interested in?

Reply:
fixed